java-security/src/test/java/com/sap/cloud/security/adapter/xs/XSUserInfoAdapterTest.java

```diff
@@ -537,28 +542,39 @@ public void canReadSystemAttributesAsPlainString() throws IOException {
 		assertThat(cut.getSystemAttribute("foo")).contains("bar");
 	}
 
-	@Test
-	public void readingEmptySystemAttributesRaisesException() throws IOException {
-		token = new XsuaaToken(IOUtils.resourceToString("/tokenWithEmptySystemPropertyAttributes.txt", UTF_8));
+	@ParameterizedTest
+	@MethodSource
+	public void readingInvalidSystemAttributesRaisesExpectedException(Token token, String attributeName, Class expectedException) {
 		cut = new XSUserInfoAdapter(token);
 
-		assertThatThrownBy(() -> cut.getSystemAttribute("foo")).isInstanceOf(XSUserInfoException.class);
+		assertThatThrownBy(() -> cut.getSystemAttribute(attributeName)).isInstanceOf(expectedException);
 	}
 
-	@Test
-	public void readingSystemAttributesAsIntegerRaisesException() throws IOException {
-		token = new XsuaaToken(IOUtils.resourceToString("/tokenWithSystemPropertyAttributeAsInteger.txt", UTF_8));
-		cut = new XSUserInfoAdapter(token);
+	private static Stream<Arguments> readingInvalidSystemAttributesRaisesExpectedException() {
+		String attributeName = "foo";
 
-		assertThatThrownBy(() -> cut.getSystemAttribute("foo")).isInstanceOf(XSUserInfoException.class);
-	}
+		Token t1 = Mockito.mock(AccessToken.class);
+		when(t1.getAttributeFromClaimAsString(XS_SYSTEM_ATTRIBUTES, attributeName)).thenReturn(null);
```

Comment:
this still is a repetitive code, it needs to be optimised plus you can also add the 2 positive test cases (canReadSystemAttributesAsStringArray, canReadSystemAttributesAsPlainString) in the same test. 
Consider initialising the token mock in the test class itself, pass expected return values of `getAttributeFromClaimAsString` and `getAttributeFromClaimAsStringList` instead of token as parameters.

Reply:
I just submitted a new version that should suit your suggestions. However, the test method contains a lot of lines to differentiate and cast between the positive and negative test cases. I think two separate positive and negative test methods would be better readable. If you agree, I can split it up while keeping the improvements in regards to less repetition.

Reply:
Thanks for the suggestion! Implemented in latest commit.